File: examples/hooks/dump-vars.lua
-- Example tangra-client deploy hook (Lua): dump every hook variable
-- into a timestamped file.
--
-- Wire up via:
--   tangra-client daemon --deploy-script-hook /path/to/dump-vars.lua
--
-- All LCM_* values are registered as globals by the hook runner
-- (see internal/hook/hook.go:registerHookContext). The `writeFile`
-- helper is also injected — it accepts (path, content) and returns
-- an error if the write fails.
--
-- NOTE: The systemd unit ships with PrivateTmp=true, which gives
-- the daemon its own /tmp namespace invisible to other processes.
-- We write under /var/log/tangra-client/ instead — /var is in
-- ReadWritePaths and the directory is visible from any shell.

exec("mkdir -p /var/log/tangra-client")

local timestamp = os.date("%Y%m%d-%H%M%S")
local out = "/var/log/tangra-client/hook-lua." .. timestamp .. ".log"

local lines = {
  "=== tangra-client deploy hook (lua) ===",
  "fired_at: " .. os.date("!%Y-%m-%dT%H:%M:%SZ"),
  "",
  "hook_type:      " .. tostring(LCM_HOOK_TYPE),
  "cert_name:      " .. tostring(LCM_CERT_NAME),
  "common_name:    " .. tostring(LCM_COMMON_NAME),
  "dns_names:      " .. tostring(LCM_DNS_NAMES),
  "ip_addresses:   " .. tostring(LCM_IP_ADDRESSES),
  "serial_number:  " .. tostring(LCM_SERIAL_NUMBER),
  "expires_at:     " .. tostring(LCM_EXPIRES_AT),
  "is_renewal:     " .. tostring(LCM_IS_RENEWAL),
  "",
  "cert_path:      " .. tostring(LCM_CERT_PATH),
  "key_path:       " .. tostring(LCM_KEY_PATH),
  "chain_path:     " .. tostring(LCM_CHAIN_PATH),
  "fullchain_path: " .. tostring(LCM_FULLCHAIN_PATH),
}

writeFile(out, table.concat(lines, "\n") .. "\n")
log("wrote " .. out)

File: examples/hooks/dump-vars.js
// Example tangra-client deploy hook (JavaScript): dump every hook
// variable into a timestamped file under /tmp.
//
// Wire up via:
//   tangra-client daemon --deploy-script-hook /path/to/dump-vars.js
//
// All LCM_* values are registered as globals by the hook runner
// (see internal/hook/hook.go:registerHookContext). The `writeFile`
// helper is also injected — it accepts (path, content) and returns
// an error if the write fails.

var d = new Date();
var pad = function (n) { return (n < 10 ? '0' : '') + n; };
var timestamp =
  d.getFullYear() + pad(d.getMonth() + 1) + pad(d.getDate()) + '-' +
  pad(d.getHours()) + pad(d.getMinutes()) + pad(d.getSeconds());

var out = '/tmp/tangra-client-hook-js.' + timestamp + '.log';

var lines = [
  '=== tangra-client deploy hook (javascript) ===',
  'fired_at: ' + d.toISOString(),
  '',
  'hook_type:      ' + LCM_HOOK_TYPE,
  'cert_name:      ' + LCM_CERT_NAME,
  'common_name:    ' + LCM_COMMON_NAME,
  'dns_names:      ' + LCM_DNS_NAMES,
  'ip_addresses:   ' + LCM_IP_ADDRESSES,
  'serial_number:  ' + LCM_SERIAL_NUMBER,
  'expires_at:     ' + LCM_EXPIRES_AT,
  'is_renewal:     ' + LCM_IS_RENEWAL,
  '',
  'cert_path:      ' + LCM_CERT_PATH,
  'key_path:       ' + LCM_KEY_PATH,
  'chain_path:     ' + LCM_CHAIN_PATH,
  'fullchain_path: ' + LCM_FULLCHAIN_PATH,
];

writeFile(out, lines.join('\n') + '\n');
log('wrote ' + out);
